packages/api/src/util/useCennznet.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019-2020 Centrality Investments Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { web3Accounts, web3Enable } from '@polkadot/extension-dapp';
+import { InjectedAccountWithMeta, MetadataDef } from '@polkadot/extension-inject/types';
+import { getSpecTypes } from '@polkadot/types-known';
+import { defaults as addressDefaults } from '@polkadot/util-crypto/address/defaults';
+
+import { ApiOptions } from '../types';
+import { Api, Api as ApiPromise } from '../Api';
+import { cennznetExtensions } from './cennznetExtensions';
+
+export async function useCennznet(
+  dAppName: string,
+  options: ApiOptions
+): Promise<{ api: ApiPromise; accounts: InjectedAccountWithMeta[] }> {
+  const api = await Api.create(options);
+  const extensions = await web3Enable(dAppName);
+  if (extensions.length === 0) {
+    return { api: api, accounts: null };
+  }
+  const polkadotExtension = extensions.find((ext) => ext.name === 'polkadot-js');
+  const metadata = polkadotExtension.metadata;
+  const checkIfMetaUpdated = localStorage.getItem(`EXTENSION_META_UPDATED`);
+  if (!checkIfMetaUpdated) {
+    const metadataDef = await extractMeta(api);
+    await metadata.provide(metadataDef);
+    localStorage.setItem(`EXTENSION_META_UPDATED`, 'true');
```

Comment:
we should allow the extension to work with different networks at different times.

similar to what we did in cennzx something like 
`cennznet-ext-meta-{api.runtimeVersion.specName}-{api.runtimeVersion.specVersion}`
should be the key.

Reply:
sweet yeah good call, updated now. 
Decided to go with network name as key

Reply:
I think better to go with specVersion as for the same network, change in specVersion will introduce new metadata..

Reply:
sweet updated now